src/NHibernate/Proxy/DynamicProxy/ProxyCacheEntry.cs

```diff
@@ -24,6 +24,7 @@ public ProxyCacheEntry(System.Type baseType, System.Type[] interfaces)
 			}
 			BaseType = baseType;
 			Interfaces = interfaces ?? new System.Type[0];
+			_uniqueInterfaces = new HashSet<System.Type>(Interfaces);
```

Comment:
There is a second HashSet used on the one I have introduced, which is now useless and could be replaced by a simple List. I will change that too.